core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4811,6 +4812,12 @@ private QueryModel rewriteSelectClause0(
             }
         }
 
+        // group-by generator can cope with virtual columns, it does not require virtual model to be its base
+        // however, sample-by single-threaded implementation still relies on the innerVirtualModel, hence the fork
+        if (useGroupByModel && sampleBy == null) {
```

Comment:
This change looks worrying to me. I'd add more tests covering things like `hour(ts1), hour(ts2)`, window functions, bind variables in keys, key/aggregate functions, and maybe something else.

Reply:
could you elaborate a little here? I am relying on the fact window function tests already exist and they did not fail. What am i missing?

Reply:
I don't think we have all edge cases in the window function test. Same with the group by tests, so the more tests we add the safer this change is.